### DISCUSSION TITLE
Cannot find class for Custom push notifications

### DISCUSSION THREAD

**Discussion by martinrehder** [ionic-team/capacitor#5667] [Q&A]
I have an existing app that already gets push notifications thru Blaze.  I need to add sound to the notifications on the android device.  

The app had the standard Braze messaging service declared in the manifest:

     <service android:name="com.braze.push.BrazeFirebaseMessagingService"
            android:exported="false">
            <intent-filter>
              <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
          </service>

But I created a new class extending the messaging service as:

      <service android:name="com.wearehathway.carlsjr.development.MyFirebaseMessagingService"
        android:exported="true">
        <intent-filter>
          <action android:name="com.google.firebase.MESSAGING_EVENT" />
        </intent-filter>
      </service>

    public class MyFirebaseMessagingService extends FirebaseMessagingService  {
      @Override
      public void onMessageReceived(RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);
        if (BrazeFirebaseMessagingService.handleBrazeRemoteMessage(this, remoteMessage)) {
          // This Remote Message originated from Braze and a push notification was displayed.
          // No further action is needed.
          Log.d("TEST", "Blaze Push received: "+remoteMessage.toString());
        } else {
          // This Remote Message did not originate from Braze.
          // No action was taken and you can safely pass this Remote Message to other handlers.
          Log.d("TEST", "NonBlaze Push received: "+remoteMessage.toString());
        }
      }
    }

This compiles and runs - until I get a push notification, then I get a fatal exception :   
"java.lang.RuntimeException: Unable to create service com.wearehathway.carlsjr.development.MyFirebaseMessagingService: java.lang.ClassNotFoundException:"

The class, MyFirebaseMessagingService is in the same folder as MainActivity - why would this class not be found?  Color me confused

**Comment by maggie-napster** [comment]
I had to add this to my service class file: 
```
package com.my.package.name;
import com.braze.push.BrazeFirebaseMessagingService;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
```

Replace `com.my.package.name` with your package name (in this case `com.wearehathway.carlsjr.development`).
